src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Listeners/ServiceBusListener.cs

```diff
@@ -2,42 +2,65 @@
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
 using System;
+using System.Collections.Generic;
+using System.Diagnostics;
 using System.Diagnostics.CodeAnalysis;
+using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Azure.ServiceBus;
 using Microsoft.Azure.ServiceBus.Core;
+using Microsoft.Azure.ServiceBus.Management;
 using Microsoft.Azure.WebJobs.Host.Executors;
 using Microsoft.Azure.WebJobs.Host.Listeners;
+using Microsoft.Azure.WebJobs.Host.Scale;
+using Microsoft.Extensions.Logging;
 
 namespace Microsoft.Azure.WebJobs.ServiceBus.Listeners
 {
-    internal sealed class ServiceBusListener : IListener
+    internal sealed class ServiceBusListener : IListener, IScaleMonitor<ServiceBusTriggerMetrics>
     {
-        private readonly MessagingProvider _messagingProvider;
+        private const string DeadLetterQueuePath = @"/$DeadLetterQueue";
+
+        private readonly MessagingProvider _messagingProvider;
+        private readonly string _functionId;
+        private readonly EntityType _entityType;
```

Comment:
@alrod had just went through and removed this EntityType I believe

Reply:
Yeah so I noticed the listener had recently been changed to be generic, but the issue is the GetMetricsAsync code changes based on whether it's a queue or a topic so the listener needs to know what EntityType it is. If there's a better way to do it I'm all ears. I thought of potentially treating everything as a queue and switching to topic if it fails, or looking at the entityPath for "/subscription/", but this seemed cleaner to me at the time

Reply:
I am ok to bring back EntityType but parsing entityPath looks also ok to me.

Reply:
Ok I think I modified it but let me know if you had something else in mind

Reply:
Looks like in passing EntityType around you've made breaking changes to public interfaces. We need to not do that unfortunately

Reply:
Ok I reverted the interface changes, but changed it so both this and the ScaleMonitor are using the same logic to parse topic/subscription.